tests/subsys/bluetooth/fast_pair/storage/factory_reset/src/main.c

```diff
@@ -0,0 +1,508 @@
+	personalized_name_validate_loaded(name);
+}
+
+static void test_store(void)
+{
+	storage_validate();
+
+	mark_test_as_finished(RESET_TEST_FINISHED_STORE_BIT_POS);
+}
+
+static void store_data_and_factory_reset(void)
+{
+	static const uint8_t seed = 50;
+	static const uint8_t key_count = ACCOUNT_KEY_MAX_CNT;
+	static const char *name = "store_data_and_factory_reset";
+	int err;
+
+	cu_account_keys_generate_and_store(seed, key_count);
+	personalized_name_store(name);
+
+	err = bt_fast_pair_factory_reset();
+	zassert_ok(err, "Unexpected error in factory reset");
+}
+
+static void test_normal_reset(void)
+{
+	store_data_and_factory_reset();
+
+	storage_validate();
+
+	mark_test_as_finished(RESET_TEST_FINISHED_NORMAL_RESET_BIT_POS);
+}
+
+static void test_interrupted_reset_begin(void)
+{
+	storage_reset_pass = true;
+
+	store_data_and_factory_reset();
+
+	/* The recent instruction should trigger reboot so this code should never be entered. */
+	ztest_test_fail();
+}
+
+static void test_interrupted_reset_finish_setup_fn(void)
+{
+	int err;
+
+	err = settings_load();
+	zassert_ok(err, "Unexpected error in settings load");
+}
+
+static void test_interrupted_reset_finish(void)
+{
+	storage_validate();
+}
+
+static void normal_test_stage_finalize(void)
+{
+	int err;
+
+	for (size_t i = 0; i < RESET_TEST_FINISHED_COUNT; i++) {
+		if (!(test_finished_mask & BIT(i))) {
+			/* Do not reboot the device, clean up and abandon the test to
+			 * properly report fail in recent tests.
+			 */
+			err = all_settings_delete();
+			__ASSERT(!err, "Unable to delete settings (err %d)", err);
+
+			return;
+		}
+	}
+
+	test_stage = RESET_TEST_STAGE_INTERRUPTED;
+	err = settings_save_one(SETTINGS_RESET_TEST_STAGE_FULL_NAME, &test_stage,
+				sizeof(test_stage));
+	__ASSERT(!err, "Unexpected error in test settings (err %d)", err);
+
+	if (IS_ENABLED(CONFIG_REBOOT)) {
+		sys_reboot(SYS_REBOOT_WARM);
+	}
+}
+
+void test_main(void)
+{
+	int err;
+
+	err = settings_subsys_init();
+	__ASSERT(!err, "Error while initializing settings subsys (err %d)", err);
+
+	err = settings_load_subtree(SETTINGS_RESET_TEST_SUBTREE_NAME);
+	__ASSERT(!err, "Error while loading test settings (err %d)", err);
+
+	__ASSERT((test_stage == RESET_TEST_STAGE_NORMAL) ||
+		 (test_stage == RESET_TEST_STAGE_INTERRUPTED),
+		 "Invalid test stage");
+
+	if (test_stage == RESET_TEST_STAGE_NORMAL) {
+		/* The test_interrupted_reset_begin test must be placed last in the suite, it
+		 * triggers sys_reboot.
+		 */
+		ztest_test_suite(fast_pair_storage_factory_reset_normal_tests,
+				 ztest_unit_test(test_section_placement),
+				 ztest_unit_test_setup_teardown(test_store, setup_fn, teardown_fn),
+				 ztest_unit_test_setup_teardown(test_normal_reset, setup_fn,
+								teardown_fn),
+				 ztest_unit_test_setup_teardown(test_interrupted_reset_begin,
+								setup_fn, unit_test_noop)
+				 );
+
+		ztest_run_test_suite(fast_pair_storage_factory_reset_normal_tests);
```

Comment:
Doesn't it return the number of failed test? So that you would no longer need to track the passed tests manually?

Reply:
This is not fully documented, so if you don't want to rely on this one, then you can stick with the current approach.

Reply:
I'll stick with the current approach.